### ISSUE TITLE
Consider remove dist variable transaction_type

### ISSUE THREAD

**Issue by Pace2Car** [apache/shardingsphere#25783]
Hi community,

The dist variable `transaction_type` is used to dynamically change the distributed transaction type. But this variable now seems to be limiting. For example, the providers in XA include atomikos and narayana, and the provider cannot be specified when setting `transaction_type`. When the user changes `transaction_type` to XA, only the default atomikos can be created. 

And after #25674, modifying the transaction rule requires rebuilding the transaction manager, so the management of the transaction rule should be contracted to the DistSQL ALTER TRANSACTION RULE.

In summary, I recommend removing dist variable `transaction_type`.

**Comment by RaigorJiang** [comment]
+1,  Use `SHOW TRANSACTION RULE` and `ALTER TRANSACTION RULE` is a better way.

**Comment by FlyingZC** [comment]
+1.
